utils/src/ooinstall/cli_installer.py

```diff
@@ -938,91 +938,10 @@ def uninstall(ctx):
-    selections = []
-    if not (latest_minor or next_major):
-        i = 0
-        if 'minor_playbook' in mapping:
-            click.echo("(%s) Update to latest %s" % (i + 1, old_version))
-            selections.append((True, False))
-            i += 1
-        if 'major_playbook' in mapping:
-            click.echo("(%s) Upgrade to next release: %s" % (i + 1, mapping['major_version']))
-            selections.append((False, True))
-            i += 1
-
-        response = click.prompt("\nChoose an option from above",
-                                type=click.Choice(list(map(str, range(1, len(selections) + 1)))))
-        latest_minor, next_major = selections[int(response) - 1]
-
-    if next_major:
-        if 'major_playbook' not in mapping:
-            click.echo("No major upgrade supported for %s %s with this version "
-                       "of atomic-openshift-utils." % (variant, old_version))
-            sys.exit(0)
-        playbook = mapping['major_playbook']
-        new_version = mapping['major_version']
-        # Update config to reflect the version we're targeting, we'll write
-        # to disk once Ansible completes successfully, not before.
-        oo_cfg.settings['variant_version'] = new_version
-        if oo_cfg.settings['variant'] == 'enterprise':
-            oo_cfg.settings['variant'] = 'openshift-enterprise'
-
-    if latest_minor:
-        if 'minor_playbook' not in mapping:
-            click.echo("No minor upgrade supported for %s %s with this version "
-                       "of atomic-openshift-utils." % (variant, old_version))
-            sys.exit(0)
-        playbook = mapping['minor_playbook']
-        new_version = old_version
-
-    click.echo("OpenShift will be upgraded from %s %s to latest %s %s on the following hosts:\n" % (
-        variant, old_version, oo_cfg.settings['variant'], new_version))
-    for host in oo_cfg.deployment.hosts:
-        click.echo("  * %s" % host.connect_to)
-
-    if not ctx.obj['unattended']:
-        # Prompt interactively to confirm:
-        if not click.confirm("\nDo you want to proceed?"):
-            click.echo("Upgrade cancelled.")
-            sys.exit(0)
-
-    retcode = openshift_ansible.run_upgrade_playbook(oo_cfg.deployment.hosts,
-                                                     playbook,
-                                                     ctx.obj['verbose'])
-    if retcode > 0:
-        click.echo("Errors encountered during upgrade, please check %s." %
-                   oo_cfg.settings['ansible_log_path'])
-    else:
-        oo_cfg.save_to_disk()
-        click.echo("Upgrade completed! Rebooting all hosts is recommended.")
+    click.echo("Upgrades are no longer supported by this version of installer")
+    click.echo("Please see the documentation for manual upgrade:")
+    click.echo("https://docs.openshift.com/container-platform/latest/install_config/upgrading/manual_upgrades.html")
```

Comment:
Lets use https://docs.openshift.com/container-platform/latest/install_config/upgrading/automated_upgrades.html instead which currently includes both quick installer based upgrades but also running ansible directly. We'll of course update that documentation to remove the quick installer bits and guide someone with how to move from quick installer to running ansible directly.

Reply:
Done